### PR TITLE
Feature issue 938 search input fields

### PR DESCRIPTION
**Pull request by ian-lin-PG**

### CODE REVIEW

File: src/components/search/search_input_component.tsx [hunk 1]
```diff
@@ -36,36 +35,38 @@ export const SearchInputComponent = (props: Props & Actions): JSX.Element => {
...
-                            debug(`SearchInputComponent location text changed to '${d}'`);
-                            setLocation(d);
-                        }}
-                        value={location}
-                        onEndEditing={(): void => props.setLocation(location)}
-                        placeholder={buildTranslatedPlaceholder(i18nRenderProp.i18n, 'Near My location')} // TODO translate
-                        placeholderTextColor={colors.white}
-                    />
-                </View>
-                <InputFormSeparator />
```
Comment (ian-lin-PG) [inline src/components/search/search_input_component.tsx]: Delete this component if no where else uses it.

File: src/components/search/search_input_component.tsx [hunk 2]
```diff
@@ -48,7 +56,7 @@ export const SearchInputComponent = (props: Props & Actions): JSX.Element => {
         debug(`SearchInput Component useEffect with '${props.currentRefinement}'`);
         props.refine(props.currentRefinement);
     }, [props.latLong]);
-
+    const [locationInput, setLocationInput]: [boolean, (b: boolean) => void] = useState();
```
Comment (ian-lin-PG) [inline src/components/search/search_input_component.tsx]: Make sure states are always initialized.

File: src/components/search/search_input_component.tsx (hunk 1 above)
Comment (ian-lin-PG) [reply src/components/search/search_input_component.tsx]: Component deleted as no other part of the code base uses it.

File: src/components/search/search_input_component.tsx (hunk 2 above)
Comment (ian-lin-PG) [reply src/components/search/search_input_component.tsx]: Renamed the state to be `hideMyLocationButton` and initialized to be `true`.

**Comment by rasmus-storjohann-PG** [comment]
One issue found:
* Open search page
* Enter some search terms and get the results
* Click on a service to open the service detail
* Click the back button
* Expected: The search terms are still there
* Observed: The search screen input fields have been reset to empty

**Pull request merged**